projects/js-upload-api/src/ClientPkey.ts

```diff
@@ -0,0 +1,209 @@
+            return true;
+        }
+        if (this.protocol !== protocol) {
+            console.debug('protocol changed', {currentProtocol: this.protocol, newProtocol: protocol}, this);
+            return true;
+        }
+        if (this.host !== host) {
+            console.debug('host changed', {currentHost: this.host, newHost: host}, this);
+            return true;
+        }
+        if (this.clientProtocolHostname !== clientProtocolHostname) {
+            console.debug('clientProtocolHostname changed', {currentClientProtocolHostname: this.clientProtocolHostname, newClientProtocolHostname: clientProtocolHostname}, this);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Get the authentication token for the current user.
+     * By default, reuses current token if available.
+     * 
+     * @param force If true, forces a new token to be generated; defaults to false
+     * 
+     * @returns The authentication token
+     * 
+     */
+    protected async getAuthToken(force = false): Promise<string> {
+        if (!force && this.authTokenValid()) {
+            return this._token || '';  // workaround ts not recognizing that _token is set
+        }
+
+        //Throw exception if invalid personalKeyId or personalKeySecret
+        if (!this.isServerConfigured()) {
+            console.debug('current config', {personalKeyId: this.personalKeyId, _personalKeySecret: this._personalKeySecret, host: this.host});
+            throw new Error('Invalid personalKeyId or personalKeySecret');
+        }
+
+        if (!force && this._getAuthTokenPromise) {
+            console.debug('reusing outstanding auth promise');
+            return await this._getAuthTokenPromise;
+        }
+
+        console.debug('getAuthToken', {personalKeyId: this.personalKeyId, personalKeySecret: this._personalKeySecret, host: this.host});
+
+        const response = await this.getToApi(
+            AUTH_API_ENDPOINT,
+            {
+                "Authorization": `PersonalKey ${this.personalKeyId}:${this._personalKeySecret}`,
+            },
+        )
+
+        const tok : string = response.token;
+        this._token = tok;
+
+        if (!this.authTokenValid()) {
+            console.error('auth token failure', {response, personalKeyId: this.personalKeyId, host: this.host});
+            throw new Error({'error': 'Auth token failure', ...(response||{})});
+        }
+
+        return tok;
+    }
+
+    /**
+     * 
+     * @param personalKeyId 
+     * @param personalKeySecret 
+     * @param org 
+     * @param protocol 
+     * @param host 
+     * @returns Promise for the authentication token
+     * 
+     * Helper to fetch a token for a given user
+     * 
+     */
+    public async fetchToken(
+        personalKeyId: string, personalKeySecret: string, org?: string, protocol = 'https', host = 'hub.graphistry.com'
```

Comment:
I see `org?`, do we need to use that somehow?

Reply:
no since we can't send the org back its not used here (yet). see https://github.com/graphistry/graphistry-js/pull/156#discussion_r1518000338. for now will mark it as unused arg by doing `_org`